doc/rm/register_tool/index.md

```diff
@@ -571,6 +571,148 @@ There is no way to guarantee that hardware is doing the right thing, but at leas
 
 Similarly it is legal to set `hwqe` true for any of these register types if the clearing wants to be monitored by the hardware outside.
 
+## Shadow Registers
+
+If in the `.hjson` file the optional key `shadowed` is set to `true` for a particular register group, the register tool will implement the corresponding registers including all fields as shadow registers using the `prim_subreg_shadow` module.
+The following section describes the concept in detail.
+
+### Overview
+
+For Comportable designs used in sensitive security environments, for example AES or HMAC peripherals, there may be a need to guard against fault injection attacks.
+One of the surfaces under threat are critical configuration settings that may be used to control keys, region access control and system security configuration.
+These registers do not typically contain secret values, but can be vulnerable to fault attacks if a fault can alter the state of a register to a specific state.
+
+The fault attacks targeting these registers can come in a few forms:
+- A fault attack directly on the storage element.
+- A fault attack on peripheral logic (for example resets) that cause the registers to revert to a default state.
+- A fault attack during the process of the write to disrupt the actual written value.
+
+Shadow registers are a mechanism to guard sensitive registers against this specific type of attack.
+
+### Description
+
+The diagram below visualizes the internals of the `prim_subreg_shadow` module used to implement shadow registers.
+
+![subreg_shadow](subreg_shadow.svg)
+
+When compared to a normal register, the `prim_subreg_shadow` module is comprised of two additional components:
+- Staged register, and
+- Shadow register.
+- The committed register is the same existing storage.
```

Comment:
I think this line should read a little cleaner, hard to parse at the moment. Maybe the whole section should be something like:

When compared to a normal register, the `prim_subreg_shadow` module adds two additional storage components, giving three storage flops in total.
The `staged register` holds a written value before being committed.
The `shadow register` holds a second copy after being committed.
And the standard storage register holds the committed value.

Reply:
Also, since there is a little bit of name collision here, perhaps we should refer to the whole thing as a `shadow register` and these storage elements as `flops`, so that would change my text above to "The `staged flop`" and "The `shadow flop`". Although now that I write it out it sounds poor. Thoughts on how to disambiguate the bigger "shadow register" (primitive) and "shadow register" (flop component)?

Reply:
Should the whole thing be called a `shadow control register` ?  then you have staged/shadow/storage register for the 3 storage elements.

Reply:
I adopted the suggestion of @sjgitty for the section.

Regarding the naming: Good point! I thought about that, too. But I didn't dare to just go ahead and change the naming. Personally, I don't like so much `shadow control register` and `shadow register`. To me, it's still too similar and with "control" I associate the use case of main control registers (maybe because I want to use this for the control register in AES...).

What about:
- primitive = "shadowed reg", copy = "shadow reg" (still too similar IMO)
- primitive = "secure reg", copy = "shadow reg"
- primitive = "shadow reg", copy = "backup/standby/redundant/duplicated reg"?

Reply:
Don't want to bikeshed here too much, so it is fine as written. We can probably clean up later, don't want this to be the reason to delay here.